### PR TITLE
#2: Added column description helper class

### LINKED ISSUE #2: Add column description helper class
### Situation

For improving the Spark schema mapping, it is useful to define a column descriptors for different types. We should add small class that represents different data types.

### Acceptance Criteria

- Column description classes are added

### PR DESCRIPTION
**Pull request by morazow**
Fixes #2

### CODE REVIEW

**Comment by redcatbear** [comment]
You should limit the checked-in Eclipse files to the formatter settings.

File: README.md [hunk 1]
```diff
@@ -1,9 +1,56 @@
...
 
+## Features
+
+- Provides a helper class to describe `JDBC` column metadata
+- Converts Exasol query column descriptions into a Spark schema
+- Generates Exasol import and export queries of cloud storage systems
+
+## Usage
+
+For example, to create column descriptions from `JDBC` query result:
```
Comment (redcatbear) [inline README.md:30]:
```suggestion
Create column descriptions from `JDBC` query result:
```

File: README.md [hunk 2]
```diff
@@ -1,9 +1,56 @@
...
+    columns.add(ColumnDescription.builder() //
+            .name(metadata.getColumnLabel(i)) //
+            .type(metadata.getColumnType(i)) //
+            .precision(metadata.getPrecision(i)) //
+            .scale(metadata.getScale(i)) //
+            .isSigned(metadata.isSigned(i)) //
+            .isNullable(metadata.isNullable(i) != columnNoNulls) //
+            .build());
+
+}
```
Comment (redcatbear) [inline README.md:46]:
```suggestion
            .isNullable(metadata.isNullable(i) != columnNoNulls) //
            .build());
}
```

`collumnNoNulls` appears in the example without introduction.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,56 @@
...
+```
+And then to generate Spark schema from column descriptions:
```
Comment (redcatbear) [inline README.md:48]:
```suggestion
Generate Spark schema from column descriptions:
```

File: doc/developer-guide/developer-guide.md
Comment (redcatbear) [inline doc/developer-guide/developer-guide.md:1]: Please add a TODO that references the issue ticket for writing the documentation.

File: src/main/java/com/exasol/spark/common/ColumnDescription.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+         */
+        public Builder name(final String name) {
+            this.name = name;
+            return this;
+        }
+
+        /**
+         * Sets the column type.
+         *
+         * Type is of a {@link java.sql.Types} {@code JDBC} types.
```
Comment (redcatbear) [inline src/main/java/com/exasol/spark/common/ColumnDescription.java:113]:
```suggestion
         * Type is one of the {@code JDBC} data types defined in {@link java.sql.Types}.
```

File: src/main/java/com/exasol/spark/common/ColumnDescription.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+         */
+        public Builder isSigned(final boolean isSigned) {
+            this.isSigned = isSigned;
+            return this;
+        }
+
+        /**
+         * Sets the boolean for nullable columns.
+         *
+         * @param isNullable boolean indicating if a column nullable or not
```
Comment (redcatbear) [inline src/main/java/com/exasol/spark/common/ColumnDescription.java:159]:
```suggestion
         * @param isNullable boolean indicating if a column is allowed to contain a {@code null} value or not
```

File: src/test/java/com/exasol/spark/common/ColumnDescriptionTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import static org.hamcrest.Matchers.startsWith;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import org.junit.jupiter.api.Test;
+
+class ColumnDescriptionTest {
+
+    @Test
```
Comment (redcatbear) [inline src/test/java/com/exasol/spark/common/ColumnDescriptionTest.java:15]:
```suggestion
class ColumnDescriptionTest {
    @Test
```

File: src/test/java/com/exasol/spark/common/ColumnDescriptionTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    void testValidateName() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/spark/common/ColumnDescriptionTest.java:16]:
```suggestion
    void testValidateNameThrowsExceptionOnMissingColumnName() {
```

File: src/test/java/com/exasol/spark/common/ColumnDescriptionTest.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        final ColumnDescription.Builder builder = ColumnDescription.builder().type(-1);
+        final IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> builder.build());
+        assertThat(exception.getMessage(), startsWith("E-SCJ-1"));
+    }
+
+    @Test
+    void testValidateType() {
```
Comment (redcatbear) [inline src/test/java/com/exasol/spark/common/ColumnDescriptionTest.java:23]:
```suggestion
    void testValidateTypeThrowsExceptionOnMissingColumnType() {
```

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: doc/developer-guide/developer-guide.md
Comment (morazow) [reply doc/developer-guide/developer-guide.md:1]: Added https://github.com/exasol/spark-connector-common-java/issues/4, not to forget

**Comment by morazow** [comment]
> You should limit the checked-in Eclipse files to the formatter settings.

Good idea, added only PK required ones. Others came from IDE settings.

**Review by redcatbear — APPROVED** [review]

**Pull request merged**